### ISSUE TITLE
Unable to create domain, load balancer, and certificate resource in a single Apply

### ISSUE THREAD

**Issue by seaburr** [digitalocean/terraform-provider-digitalocean#587]
### Terraform Version
Terraform v0.14.6
+ provider registry.terraform.io/digitalocean/digitalocean v2.5.1

### Affected Resources
- digitalocean_certificate
- digitalocean_loadbalancer
- digitalocean_domain

*NOTE:* I have not tested other versions of Terraform so I am not certain if this is an issue with Terraform core.

### Terraform Configuration Files
```hcl
terraform {
    required_providers {
        digitalocean = {
            source = "digitalocean/digitalocean"
            version = "2.5.1"
        }
    }
}

variable "do_token" {
    description = "DigitalOcean API access token."
}

variable "domain" {
  description = "Domain name."
}

variable "digitalocean_region" {
  description = "DO region where resources should be deployed."
  default = "nyc1"
}

resource "digitalocean_domain" "web-domain" {
   name = var.domain
   ip_address = digitalocean_loadbalancer.loadbalancer.ip
}

resource "digitalocean_record" "CNAME-www" {
  domain = digitalocean_domain.web-domain.id
  type = "CNAME"
  name = "www"
  value = "@"
}

resource "digitalocean_certificate" "ssl-cert" {
  name    = "le-ssl-cert"
  type    = "lets_encrypt"
  domains = [ var.domain ]

  lifecycle {
    create_before_destroy = true
  }
}

resource "digitalocean_loadbalancer" "loadbalancer" {
  name = "loadbalancer"
  region = var.digitalocean_region
  droplet_tag = "web-nodes"

  forwarding_rule {
    entry_port = 443
    entry_protocol = "https"

    target_port = 80
    target_protocol = "http"

    certificate_name = digitalocean_certificate.ssl-cert.name
  }

  healthcheck {
    port = 22
    protocol = "tcp"
  }
}

resource "digitalocean_droplet" "test-droplet" {
  image = "ubuntu-18-04-x64"
  name = "test-droplet"
  region = "nyc1"
  size = "s-1vcpu-1gb"
  private_networking = true
}
```

*NOTE:* See makefile attached with below gist as it might make replication easier.

### Debug Output
https://gist.github.com/seaburr/676bf963f2ee77b6242251a101e9ff14

### Expected Behavior
Terraform apply should not fail. The domain resource should be created first, then the certificate, then the load balancer.

### Actual Behavior
Terraform apply fails while trying to create the certificate because the domain resource does not exist.

### Steps to Reproduce
1. terraform init
2. terraform apply

### Important Factoids
I found a simple workaround to this by not managing the domain resource under Terraform but this is less than ideal because it results in multiple A records being added on the domain with each subsequent destroy and apply cycle. Additionally, importing the domain does not work. You are able to destroy but on the next apply, the same issue arises where resources are created out of order.

### References
Originally, I created a ticket with DigitalOcean support on this issue which may have some useful additional details from our discussion (Ticket #4464011).

**Comment by andrewsomething** [comment]
To be able to create the certificate using Let's Encrypt, the domain needs to be managed by DigitalOcean. In the config you shared, there is no direct dependency between the domain and the certificate. So Terraform may attempt to create the certificate before the domain exists.

Here's a config that works for me with one way of creating that dependency. 

```hcl
resource "digitalocean_domain" "web-domain" {
   name = var.domain
}

resource "digitalocean_record" "A" {
  domain = digitalocean_domain.web-domain.id
  type = "A"
  name = "@"
  value = digitalocean_loadbalancer.loadbalancer.ip
}

resource "digitalocean_record" "CNAME-www" {
  domain = digitalocean_domain.web-domain.id
  type = "CNAME"
  name = "www"
  value = "@"
}

resource "digitalocean_certificate" "ssl-cert" {
  name    = "le-ssl-cert"
  type    = "lets_encrypt"
  domains = [ digitalocean_domain.web-domain.name ]

  lifecycle {
    create_before_destroy = true
  }
}

resource "digitalocean_loadbalancer" "loadbalancer" {
  name = "loadbalancer"
  region = var.digitalocean_region
  droplet_tag = "web-nodes"

  forwarding_rule {
    entry_port = 443
    entry_protocol = "https"

    target_port = 80
    target_protocol = "http"

    certificate_name = digitalocean_certificate.ssl-cert.name
  }

  healthcheck {
    port = 22
    protocol = "tcp"
  }
}

resource "digitalocean_droplet" "test-droplet" {
  image = "ubuntu-18-04-x64"
  name = "test-droplet"
  region = "nyc1"
  size = "s-1vcpu-1gb"
  private_networking = true
}
```

**Comment by seaburr** [comment]
Thank you. It looks like the fix is really in changing this:

```hcl
resource "digitalocean_domain" "web-domain" {
   name = var.domain
   ip_address = digitalocean_loadbalancer.loadbalancer.ip
}
```

To this:
```hcl
resource "digitalocean_domain" "web-domain" {
   name = var.domain
}
```

Which removes the circular dependency. I should have noticed this when I tried to to add `depends_on = digitalocean_domain` to the load balancer and certificate resource.

Thank you for helping me understand what was going on here and providing an example of a fix. I was able to verify the resource creation order looks good. I cannot completely confirm this works because the certificates API is returning errors (403 and now 500).
